### PR TITLE
feat: add email / social info to `useAppKitAccount` hook

### PR DESCRIPTION
**Pull request by magiziz**
# Description

Added email, username and some other auth info to `useAppKitAccount` hook

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-1410

# Showcase (Optional)

Screenshot 2024-12-30 at 15 38 20

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [x] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/appkit/src/client.ts [hunk 1]
```diff
@@ -1223,6 +1240,9 @@ export class AppKit {
       await provider.getSmartAccountEnabledNetworks()
       this.setLoading(false)
     })
+    provider.onSocialConnected(({ userName }) => {
+      this.setUsername(userName, ChainController.state.activeChain as ChainNamespace)
+    })
```
Comment (magiziz) [inline packages/appkit/src/client.ts:1245]: We can only get the username when connecting to a social. This means we'd have to store the username in local storage like we currently do and use that value in our `AccountController.state.username` state.

Magic doesn't give us back username when calling `this.magic.getUser()` in secure site

File: apps/laboratory/src/components/AppKitInfo.tsx [hunk 2]
```diff
@@ -55,6 +59,42 @@ export function AppKitInfo({ caipAddress, address, chainId, clientId }: AppKitIn
...
+          )}
+
+          <Box>
+            <Heading size="xs" textTransform="uppercase" pb="2">
+              Smart Account Status
+            </Heading>
+            <Text data-testid="w3m-sa-account-status">
+              {isSmartAccountDeployed ? 'Deployed' : 'Not Deployed'}
+            </Text>
+          </Box>
```
Comment (tomiir) [inline apps/laboratory/src/components/AppKitInfo.tsx:97]: maybe nit but can we make a component for this?  like `InfoRow` or something

File: packages/appkit/src/client.ts (hunk 1 above)
Comment (tomiir) [reply packages/appkit/src/client.ts:1245]: wym? wouldn't `connect` have the username when reconnecting?

File: packages/appkit/src/client.ts [hunk 3]
```diff
@@ -490,6 +496,14 @@ export class AppKit {
     AccountController.setProfileImage(profileImage, chain)
   }
 
+  public setEmail: (typeof AccountController)['setEmail'] = (email, chain) => {
+    AccountController.setEmail(email, chain)
+  }
+
+  public setUsername: (typeof AccountController)['setUsername'] = (username, chain) => {
+    AccountController.setUsername(username, chain)
```
Comment (tomiir) [inline packages/appkit/src/client.ts:504]: this is only for social accounts right? wouldn't it make sense to store the user object directly?

File: packages/core/exports/react.ts [hunk 4]
```diff
@@ -20,14 +20,23 @@ export function useAppKitNetworkCore(): Pick<
...
+
   const { activeCaipAddress } = useSnapshot(ChainController.state)
 
   return {
     caipAddress: activeCaipAddress,
     address: CoreHelperUtil.getPlainAddress(activeCaipAddress),
     isConnected: Boolean(activeCaipAddress),
-    status
+    status,
+    user: {
```
Comment (tomiir) [inline packages/core/exports/react.ts:34]: shouldn't this be part of wallet info instead? it's not really user data but rather wallet data which has user info from auth/email/SA

Comment (tomiir) [reply packages/core/exports/react.ts:34]: nvm

Comment (tomiir) [inline packages/core/exports/react.ts:34]: I'd call this embedded wallet info or something like that

File: packages/core/exports/vue.ts [hunk 5]
```diff
@@ -23,9 +29,29 @@ export function useAppKitAccount() {
...
+    state.value.user.accountType = val ?? null
+  })
+
+  const unsubscribeEmail = AccountController.subscribeKey('email', val => {
+    state.value.user.email = val ?? null
+  })
+
+  const unsubscribeUsername = AccountController.subscribeKey('username', val => {
+    state.value.user.username = val ?? null
+  })
```
Comment (tomiir) [inline packages/core/exports/vue.ts:46]: they are all updated at the same time (on connection) so I'd keep a single object for this and we serialize all data as is to the hook. wyt?

File: packages/core/src/controllers/AccountController.ts [hunk 6]
```diff
@@ -144,6 +146,14 @@ export const AccountController = {
     ChainController.setAccountProp('profileImage', profileImage, chain)
   },
 
+  setEmail(email: AccountControllerState['email'], chain: ChainNamespace) {
+    ChainController.setAccountProp('email', email, chain)
+  },
+
+  setUsername(username: AccountControllerState['username'], chain?: ChainNamespace) {
+    ChainController.setAccountProp('username', username, chain)
+  },
```
Comment (tomiir) [inline packages/core/src/controllers/AccountController.ts:155]: Tricky case, cause they should be updated on all namespaces but not all namespaces might support auth wallet

File: packages/appkit/src/client.ts (hunk 1 above)
Comment (magiziz) [reply packages/appkit/src/client.ts:1245]: For some reason it doesn't expose username. It only exposes username when connecting to a social

File: apps/laboratory/src/components/AppKitInfo.tsx (hunk 2 above)
Comment (magiziz) [reply apps/laboratory/src/components/AppKitInfo.tsx:97]: good idea

File: packages/core/exports/vue.ts (hunk 5 above)
Comment (magiziz) [reply packages/core/exports/vue.ts:46]: yeap let me do that

**Review by tomiir — APPROVED** [review]

**Pull request merged**